Kiosk/Views/InkRecognizerExplorer/DualCanvas.xaml.cs

```diff
@@ -0,0 +1,860 @@
+
+            float bottomRightX = (float)recoUnit.rotatedBoundingRectangle[2].x;
+            float bottomRightY = (float)recoUnit.rotatedBoundingRectangle[2].y;
+
+            float bottomLeftX = (float)recoUnit.rotatedBoundingRectangle[3].x;
+            float bottomLeftY = (float)recoUnit.rotatedBoundingRectangle[3].y;
+
+            // Height and width of bounding rectangle to get font size and width for text layout
+            float height = GetDistanceBetweenPoints(topRightX, bottomRightX, topRightY, bottomRightY) * dipsPerMm;
+            float width = GetDistanceBetweenPoints(bottomLeftX, bottomRightX, bottomLeftY, bottomRightY) * dipsPerMm;
+
+            if (height < 45)
+            {
+                height = 45;
+            }
+
+            // Transform to get correct angle of text
+            float centerPointX = ((topLeftX + topRightX) / 2) * dipsPerMm;
+            float centerPointY = ((topLeftY + bottomLeftY) / 2) * dipsPerMm;
+            var centerPoint = new Vector2(centerPointX, centerPointY);
+
+            Matrix3x2 angle = GetRotationAngle(bottomLeftX, bottomRightX, bottomLeftY, bottomRightY, centerPoint);
+
+            args.DrawingSession.Transform = angle;
+
+            var textFormat = new CanvasTextFormat()
+            {
+                FontSize = height - 5,
+                WordWrapping = CanvasWordWrapping.NoWrap,
+                FontFamily = "Ink Free"
+            };
+
+
+            // Build string to be drawn to canvas
+            string textLine = string.Empty;
+            foreach (var item in childIds)
+            {
+                int id = int.Parse(item.ToString());
+                (string text, _) = recoText[id];
+
+                textLine += text + " ";
+            }
+
+            var textLayout = new CanvasTextLayout(sender.Device, textLine, textFormat, width, height);
+
+            // Associate correct color with each word in string
+            int index = 0;
+            foreach (var item in childIds)
+            {
+                int id = int.Parse(item.ToString());
+                (string text, Color color) = recoText[id];
+
+                textLayout.SetColor(index, text.Length, color);
+
+                index += text.Length + 1;
+            }
+
+            args.DrawingSession.DrawTextLayout(textLayout, floatX * dipsPerMm, floatY * dipsPerMm, Colors.Black);
+            args.DrawingSession.Transform = initialTransformation;
+        }
+
+        private void DrawShape(InkRecognitionUnit recoUnit, CanvasDrawEventArgs args)
+        {
+            string recognizedObject = recoUnit.recognizedObject;
+            switch (recognizedObject)
+            {
+                case "circle":
+                    DrawCircle(recoUnit, args);
+                    break;
+                case "ellipse":
+                    DrawEllipse(recoUnit, args);
+                    break;
+                case "drawing":
+                    DrawLine(recoUnit, args);
```

Comment:
is this accurate?  you might check with Yibo on this but I thought drawing came back if it was any unknown collection of strokes that aren't text so us being able to redraw it in a "neat" way would be impossible.  looking at the DrawLine function it's only drawing a line as the method name suggests so i'd think this would not do the right thing for some cases.

Reply:
It was agreed on that users would most likely try to underline text or draw lines so I added this method in there to draw horizontal lines. The service does not recognize lines like this but I checked the drawing bounding boxes and if they were less than a certain height and at least a certain width I would us that to draw them. If a line is inked in a fairly straight and horizontal manner it will be rendered, but if a line is say a squiggle or something weird nothing will be rendered on the second canvas.

Reply:
I can always rename the method to something a little more clear like DrawHorizontalLine and document it better with a comment explaining lines do not come back with any data about their strokes.